### PR TITLE
Issue 5162 - Lib389 - verify certificate type before adding

### PR DESCRIPTION
**Pull request by mreynolds389**
Description: 

Verify that when importing a certificate that is the correct type.  Also cleanup temporary certs that are created when processing a bundle of certs in a PEM file.

relates: https://github.com/389ds/389-ds-base/issues/5162

### CODE REVIEW

File: rpm/389-ds-base.spec.in [hunk 1]
```diff
@@ -386,6 +386,7 @@ Requires: python%{python3_pkgversion}-dateutil
 Requires: python%{python3_pkgversion}-argcomplete
 Requires: python%{python3_pkgversion}-libselinux
 Requires: python%{python3_pkgversion}-setuptools
+Requires: python%{python3_pkgversion}-cryptography
```
Comment (droideck) [inline rpm/389-ds-base.spec.in:390]: I think, we also need to add the dependency here:
https://github.com/389ds/389-ds-base/blob/main/src/lib389/requirements.txt

And here:
https://github.com/389ds/389-ds-base/blob/main/src/lib389/setup.py#L94

File: src/lib389/lib389/nss_ssl.py [hunk 2]
```diff
@@ -1271,6 +1277,14 @@ def add_ca_cert_bundle(self, cert_file, nicknames):
...
                                 if names_len > ca_count:
                                     if nicknames[ca_count].lower() == CERT_NAME.lower() or nicknames[ca_count].lower() == CA_NAME.lower():
+                                        # cleanup
+                                        try:
+                                            for tmp_file in ca_files_to_cleanup:
+                                                os.remove(tmp_file)
+                                        except IOError as e:
+                                            # failed to remove tmp cert
+                                            log.debug("Failed to remove tmp cert file: " + str(e))
+                                            pass
```
Comment (droideck) [inline src/lib389/lib389/nss_ssl.py]: I don't think `pass` is needed here as we have the `log.debug` command and there's no need in a placeholder statement.

File: src/lib389/lib389/nss_ssl.py [hunk 3]
```diff
@@ -1290,21 +1304,39 @@ def add_ca_cert_bundle(self, cert_file, nicknames):
...
                                 log.info(f"Successfully added CA certificate ({ca_cert_name})")
+
+                    # All done, cleanup
+                    try:
+                        for tmp_file in ca_files_to_cleanup:
+                            os.remove(tmp_file)
+                    except IOError as e:
+                        # failed to remove tmp cert
+                        log.debug("Failed to remove tmp cert file: " + str(e))
+                        pass
```
Comment (droideck) [inline src/lib389/lib389/nss_ssl.py]: Same here

File: src/lib389/lib389/nss_ssl.py [hunk 4]
```diff
@@ -1290,21 +1304,39 @@ def add_ca_cert_bundle(self, cert_file, nicknames):
...
                     ca_file.close()
-                    for tmp_file in ca_file_to_cleanup:
-                        os.remove(tmp_file)
+                    try:
+                        for tmp_file in ca_files_to_cleanup:
+                            os.remove(tmp_file)
+                    except IOError as e:
+                        # failed to remove tmp cert
+                        log.debug("Failed to remove tmp cert file: " + str(e))
+                        pass
```
Comment (droideck) [inline src/lib389/lib389/nss_ssl.py]: And here

**Review by droideck — CHANGES_REQUESTED** [review]

File: rpm/389-ds-base.spec.in (hunk 1 above)
Comment (mreynolds389) [reply rpm/389-ds-base.spec.in:390]: Well it's not an install requirement, but a runtime requirement.  I'm adding it anyway, but I'm not sure we actually need it in setup.py

Comment (droideck) [reply rpm/389-ds-base.spec.in:390]: Yeah, it's kind of a misleading naming from their side.

`install_requires` actually declares the package's core dependencies, without which it won’t be able to run: https://packaging.python.org/en/latest/discussions/install-requires-vs-requirements/

But also, we actually need the package in specfile's `BuildRequires`. It fails without it when I try with Mock.

**Review by droideck — APPROVED** [review]
Besides the minor issue, looks good!

File: rpm/389-ds-base.spec.in (hunk 1 above)
Comment (mreynolds389) [reply rpm/389-ds-base.spec.in:390]: Fixed

**Pull request merged**